Add Adversus Actions: Create/Update Lead, Add Note/Activity, Change Lead Status, Assign to Campaign

harshasiddartha opened the pull request:
## WHY

These actions address common use cases for Pipedream users integrating with Adversus. Currently, the Adversus component only has basic app configuration, but lacks the essential actions needed for lead management workflows. Adding these actions enables users to:

- **Automate lead lifecycle management**: Create, update, and manage leads programmatically without manual intervention
- **Enhance CRM integration workflows**: Sync lead data between Adversus and other systems (Salesforce, HubSpot, etc.)
- **Streamline sales operations**: Automatically assign leads to campaigns and track status changes
- **Improve team collaboration**: Add notes and activities to leads as part of automated workflows

These actions fill a critical gap in the Adversus integration, allowing users to build comprehensive automation workflows that weren't previously possible.

## WHAT

This PR adds four new actions to the Adversus component:

- **Create or Update Lead** - Create a new lead or update an existing lead
- **Add Note or Activity** - Add a note or activity to a lead
- **Change Lead Status** - Change the status of a lead
- **Assign Lead to Campaign** - Assign a lead to a campaign

All actions include API documentation references to https://solutions.adversus.io/api

## Changes Made

### App File Updates (`adversus.app.mjs`)
- Added proper authentication methods (Basic Auth)
- Added API methods for all actions:
  - `createLead()` - Create a new lead
  - `updateLead()` - Update an existing lead
  - `addNoteToLead()` - Add a note to a lead
  - `addActivityToLead()` - Add an activity to a lead
  - `changeLeadStatus()` - Change lead status
  - `assignLeadToCampaign()` - Assign lead to campaign
- Added prop definitions for reusable components (leadId, campaignId, statusId)
- Fixed data merging in `changeLeadStatus` and `assignLeadToCampaign` methods
- Added JSDoc comments to all API methods for better documentation

### New Actions
- All actions include proper annotations (destructiveHint, openWorldHint, readOnlyHint)
- All actions have clear descriptions with API documentation links
- Proper error handling and validation
- Consistent use of propDefinitions for reusable props (including leadId in create-or-update-lead)

### Documentation
- Updated README.md with API documentation link

## Testing
- All code follows Pipedream component patterns
- No linting errors
- Proper error handling implemented
- JSDoc comments added for all methods (meets 80% coverage requirement)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added four Adversus actions: add note & activity to leads, assign leads to campaigns, change lead status, and create-or-update leads.
  * Exposed integration properties for Lead ID, Campaign ID, and Status ID for easier configuration.
* **Documentation**
  * Expanded Adversus docs with API reference and example use cases (real-time notifications for high‑value deals, performance analytics reporting).
* **Chores**
  * Bumped Adversus component version to 0.7.0.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

harshasiddartha commented:
@coderabbitai please re-scan the PR. All docstrings have been added with JSDoc format to all methods in adversus.app.mjs and all run methods in the action files.

harshasiddartha commented:
@coderabbitai please review the latest changes. All docstrings have been added and the PR has been updated with a single clean commit.

jcortes commented on components/adversus/adversus.app.mjs:
```diff
@@ -1,11 +1,154 @@
+    _auth() {
+      return {
+        username: this.$auth.username || this.$auth.api_key,
+        password: this.$auth.password || this.$auth.api_secret,
```
Since only these inputs are available
```suggestion
        username: this.$auth.username,
        password: this.$auth.password,
```

jcortes reviewed (changes requested):
Hi @harshasiddartha please increase the version of `package.json` to `"version": "0.7.0",` since you are adding new components to the app. I've just left a couple of suggestions which are options other than that it's looking good thanks!

harshasiddartha commented:
Hi @jcortes! Thank you for the review feedback. I've addressed all your suggestions:

✅ **Updated package.json version** to `0.7.0` as requested

✅ **Simplified `_auth()` method** - Removed the fallbacks to `api_key` and `api_secret`, now only uses `username` and `password` directly

✅ **Simplified `_makeRequest()` method** - Now directly uses `this._auth()` instead of storing it in a variable first

All changes have been committed and pushed. The PR is ready for your review!

jcortes reviewed (commented):
Hi @harshasiddartha lgtm! Ready for QA!
